contracts/libraries/VaultLifecycleYearn.sol

```diff
@@ -0,0 +1,554 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.3;
+pragma experimental ABIEncoderV2;
+
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import {Vault} from "./Vault.sol";
+import {IYearnVault} from "../interfaces/IYearn.sol";
+import {
+    IStrikeSelection,
+    IOptionsPremiumPricer
+} from "../interfaces/IRibbon.sol";
+import {GnosisAuction} from "./GnosisAuction.sol";
+import {
+    IOtokenFactory,
+    IOtoken,
+    IController,
+    GammaTypes
+} from "../interfaces/GammaInterface.sol";
+import {IERC20Detailed} from "../interfaces/IERC20Detailed.sol";
+
+library VaultLifecycleYearn {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    struct CloseParams {
+        address OTOKEN_FACTORY;
+        address USDC;
+        address currentOption;
+        uint256 delay;
+        uint16 lastStrikeOverride;
+        uint256 overriddenStrikePrice;
+    }
+
+    function commitAndClose(
+        address strikeSelection,
+        address optionsPremiumPricer,
+        uint256 premiumDiscount,
+        CloseParams calldata closeParams,
+        Vault.VaultParams calldata vaultParams,
+        Vault.VaultState calldata vaultState,
+        address collateralAsset
+    )
+        external
+        returns (
+            address otokenAddress,
+            uint256 premium,
+            uint256 strikePrice,
+            uint256 delta
+        )
+    {
+        uint256 expiry;
+
+        // uninitialized state
+        if (closeParams.currentOption <= address(1)) {
```

Comment:
I [changed this recently](https://github.com/ribbon-finance/ribbon-v2/blob/chainsafe-audit/contracts/libraries/VaultLifecycle.sol#L51), needs to be `if (closeParams.currentOption == address(0)) {`